src/components/screens/vacation-requests/vacation-requests-table/vacation-requests-table-toolbar/toolbar-form/toolbar-form.tsx

```diff
@@ -0,0 +1,173 @@
+  selectedRowIds: GridRowId[] | undefined;
+  rows: DataGridRow[] | undefined;
+  toolbarFormMode: ToolbarFormModes;
+  setToolbarFormMode: Dispatch<SetStateAction<ToolbarFormModes>>;
+}
+/**
+ * Table form component
+ *
+ * @param props TableFormProps
+ */
+const TableForm = (props: TableFormProps) => {
+  const {
+    formOpen,
+    setFormOpen,
+    vacationRequestStatuses,
+    setVacationRequestStatuses,
+    vacationRequests,
+    setVacationRequests,
+    selectedRowIds,
+    rows,
+    toolbarFormMode,
+    setToolbarFormMode
+  } = props;
+  const dateTimeNow = DateTime.now();
+  const [initialStartDate, setInitialStartDate] = useState<DateTime | undefined>(dateTimeNow);
+  const [initialEndDate, setInitialEndDate] = useState<DateTime | undefined>(dateTimeNow);
+  const [vacationData, setVacationData] = useState<VacationData>({
+    startDate: dateTimeNow,
+    endDate: dateTimeNow,
+    type: VacationType.VACATION,
+    message: "",
+    days: 1
+  });
+  const [readyToSubmit, setReadyToSubmit] = useState(false);
+  const [selectedVacationRequestId, setSelectedVacationRequestId] = useState<string | undefined>(
+    undefined
+  );
+  const { createVacationRequest } = CreateVacationRequest({
+    vacationRequests: vacationRequests,
+    vacationRequestStatuses: vacationRequestStatuses,
+    setVacationRequestStatuses: setVacationRequestStatuses,
+    setVacationRequests: setVacationRequests
+  });
+  const { updateVacationRequest } = UpdateVacationRequest({
+    vacationRequests: vacationRequests,
+    setVacationRequests: setVacationRequests
+  });
+
+  /**
+   * Reset vacation data
+   */
+  const resetVacationData = () => {
+    setVacationData({
+      type: "VACATION",
+      message: "",
+      startDate: dateTimeNow,
+      endDate: dateTimeNow,
+      days: 1
+    });
+    setInitialStartDate(dateTimeNow);
+    setInitialEndDate(dateTimeNow);
+  };
+
+  /**
+   * Determine toolbar form mode
+   */
+  useEffect(() => {
+    determineToolbarFormMode({
+      formOpen: formOpen,
+      selectedRowIds: selectedRowIds,
+      setToolbarFormMode: setToolbarFormMode
+    });
+  }, [selectedRowIds, formOpen]);
+
+  /**
+   * Set vacation data from selected row if toolbar is in edit mode
+   */
+  useEffect(() => {
+    if (toolbarFormMode === ToolbarFormModes.EDIT && selectedRowIds?.length && rows?.length) {
+      getVacationDataFromRow({
+        rows: rows,
+        selectedRowIds: selectedRowIds,
+        vacationRequests: vacationRequests,
+        setSelectedVacationRequestId: setSelectedVacationRequestId,
+        setInitialStartDate: setInitialStartDate,
+        setInitialEndDate: setInitialEndDate,
+        setVacationData: setVacationData
+      });
+    } else {
+      resetVacationData();
+    }
+  }, [toolbarFormMode]);
+
+  /**
+   * Vacation data validation
+   */
+  useEffect(() => {
+    if (hasAllPropsDefined(vacationData)) {
+      setReadyToSubmit(true);
+    } else {
+      setReadyToSubmit(false);
+    }
+  });
```

Comment:
Does this readyToSubmit boolean actually control the submission of the form? I don't see any other references to it except here.

Reply:
Nevermind, I just missed it

Reply:
However, could the validator function be directly used to control the button? <Button disabled={hasAllPropsDefined(vacationData))}> instead of <Button disabled={readyToSubmit}>

Reply:
Yes, that works too and is less code. Committed.